mysys/buffered_error_log.cc

```diff
@@ -0,0 +1,71 @@
+
+#include "buffered_error_log.h"
+
+
+
+Buffered_error_logger buffered_error_log;
+char *buffered_error_log_filename = nullptr;
+
+void Buffered_error_logger::resize(std::size_t buffer_size) {
+  std::lock_guard<std::mutex> lk{data_mtx};
+
+  if (data.get() != nullptr && buffer_size == data->capacity()) {
+    return;
+  }
+
+  // Write out what we have currently, that way we don't have
+  // to deal with old data in the buffer
+  write_to_disk_();
+
+  if (buffer_size == 0) {
+    data.reset();
+    return;
+  }
+
+  data_t new_buffer(new std::string());
+  new_buffer->reserve(buffer_size);
+  data.swap(new_buffer);
+}
+
+Buffered_error_logger::~Buffered_error_logger() { write_to_disk(); }
+
+void Buffered_error_logger::log(const char *msg, size_t len) {
+  std::lock_guard<std::mutex> lk{data_mtx};
+  if (data.get() == nullptr || data->capacity() == 0) return;
```

Comment:
when do we come log and the buffer capacity is null or 0?

Reply:
If `log_to_buffered_error_log` is true, but the user didn't configure buffered logging.